src/video/SDL_video.c

```diff
@@ -1246,6 +1246,8 @@ SDL_DisplayID SDL_GetDisplayForWindowCoordinate(int coordinate)
 SDL_DisplayID SDL_GetDisplayForWindow(SDL_Window *window)
 {
     SDL_DisplayID displayID = 0;
+    const int win_x = (window->flags & SDL_WINDOW_FULLSCREEN) ? window->x : window->windowed.x;
```

Comment:
Why not always use window->x/y? They're intended to always be current, whereas windows.x is the application requested windowed position.

Reply:
The point functions were always using the windowed coordinates before, so I didn't want to change them in the windowed case as I wasn't sure if there was a reason for them being that way.  Since there isn't, I'll change them to use the window x/y.